### DISCUSSION TITLE
`onSelect` for menu item?

### DISCUSSION THREAD

**Discussion by rin-yato** [chakra-ui/zag#1540] [Ideas]
I'm not quite sure how to handle selection for menu items. Currently I have a `handleSelection` on the root that just run a switch statement on the item's value and deciding what to do after that. 

I was hoping that there would be a simpler way. Would it be possible to have an `onSelect` callback for menu item where it would be trigger on click, touch or keyboard action.

**Comment by cschroeter** [comment]
@segunadebayo 

What are your thoughts on this?

**Reply by segunadebayo** [reply]
We could consider dispatching a custom event on the menu item itself that the user can listen to.

**Comment by anubra266** [comment]
@rin-yato 
https://zagjs.com/components/react/menu#:~:text=menu%20item%20changes.-,onSelect,Function%20called%20when%20a%20menu%20item%20is%20selected.,-anchorPoint
